packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -347,67 +347,32 @@ const positiveNegativeNumberCellView = PositiveNegativeNumberCellView.compose({
 });
 ```
 
-Finally, here is a column element that allows a user to register a callback for a click event on a button inside the cell template:
+### Event Delegation
 
-```TS
-type TableColumnButtonCellData = TableStringField<'id'>;
-
-public class TableColumnButton extends TableColumn<TableColumnButtonCellData> {
-    ...
-
-    public cellRecordFieldNames = ['id'] as const;
+Because the cell view is not a descendant of the column element, we must add special support for the column to handle events that originate in the cells. A column will define a `columnDelegatedEvents` property which is an array of event names that should be delegated.
 
-    @attr
-    public idKey: string;
-
-    public getDataRecordFieldNames(): string[] {
-        return [valueKey];
-    }
-
-    public cellViewTag = 'table-cell-view-button';
-
-    public validateCellData(cellData: TCellRecord): void {
-        if (typeof(cellData['id']) !== 'string') {
-            throw new Error('Type for cellData is incorrect!');
-        }
-    }
+```TS
+TableColumn {
+   abstract readonly columnDelegatedEvents: readonly string[];
 }
 
-class ButtonCellView extends TableCellView<TableColumnButtonCellData> {
-    public onButtonClick(): void {
-        this.$emit('button-click', { data: this.cellRecord.id })
-    }
+AnchorTableColumn extends TableColumn {
+   readonly columnDelegatedEvents = ['click'] as const;
 }
-const buttonCellView = ButtonCellView.compose({
-    baseName: 'table-cell-view-button',
-    template: html<ButtonCellView>`
-        <nimble-button @click="${(x) => x.onButtonClick()}">
-            <span>Press Me</span>
-        </nimble-button>`,
-    styles: /* styling */
-});
-```
-
-Angular template:
-
-```HTML
-<nimble-table (button-click)="doSomething($event)">
-    <nimble-table-column-button></nimble-table-column-button>
-</nimble-table>
 ```
 
-Angular code:
+Upon connecting a cell view to the DOM, we will attach an event listener for each event type in `columnDelegatedEvents`. The handler will wrap the original event in a new `CustomEvent` and dispatch that to the column. It will also stop propagation on the original event, and prevent default if the `dispatchEvent` call returns `false`.
 
 ```TS
-private doSomething(CustomEvent e): void {
-    ...
-}
-
+this.addEventListener(delegatedEventName, event => {
+    event.stopPropagation();
+    if (!this.column.dispatchEvent(new CustomEvent('delegated-event', { details: { event }}))) {
```

Comment:
1. Can you spell out the type of the details object itself? What properties will it have; document it similar to the selection api `selection-change` event [TableRowSelectionEventDetail ](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table/specs/table-row-selection.md#api)
2. I called out also including the event name because I do not believe that is part of the event object. Part of the api is registering multiple events to listen for, can the user distinguish them without the name?
3. I'd also encourage a more descriptive name for the event that is being delegated, I can imagine it being confusing in an event hander to do:
    
    ```ts
    addEventListener('delegated-event', event => {
        event.details.event.doSomething();
    })
    ```

I think the explicit example would make it clearer that a more descriptive name would be helpful, i.e. `originalEvent` or `delegatedEvent`.

Reply:
1. Done
2. The event name is `Event.type`, so they can distinguish using that. I think it would be redundant to add the name to the details separately.
3. Changed to `originalEvent`